use_cases/eluc/prescriptors/nsga2/torch_prescriptor.py

```diff
@@ -0,0 +1,264 @@
+
+        self.eval_df = eval_df
+        self.encoded_eval_df = encoder.encode_as_df(eval_df)
+        self.encoder = encoder
+        context_ds = TorchDataset(self.encoded_eval_df[constants.CAO_MAPPING["context"]].to_numpy(),
+                                  np.zeros((len(self.encoded_eval_df), len(constants.RECO_COLS))))
+        self.context_dl = torch.utils.data.DataLoader(context_ds, batch_size=batch_size, shuffle=False)
+        self.batch_size = batch_size
+
+        self.predictor = predictor
+
+    def _reco_tensor_to_df(self, reco_tensor: torch.Tensor, context_df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Converts raw Candidate neural network output tensor to scaled dataframe.
+        Sets the indices of the recommendations so that we can subtract from the context to get
+        the land diffs.
+        """
+        reco_df = pd.DataFrame(reco_tensor.cpu().numpy(), index=context_df.index, columns=constants.RECO_COLS)
+        reco_df = reco_df.clip(0, None) # ReLU
+        reco_df[reco_df.sum(axis=1) == 0] = 1 # Rows of all 0s are set to 1s
+        reco_df = reco_df.div(reco_df.sum(axis=1), axis=0) # Normalize to sum to 1
+        reco_df = reco_df.mul(context_df[constants.RECO_COLS].sum(axis=1), axis=0) # Rescale to match original sum
+        return reco_df
+
+    def _reco_to_context_actions(self, reco_df: pd.DataFrame, context_df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Converts recommendation df and original context df to context + actions df.
+        Uses original context to compute diffs based on recommendations - original context.
+        """
+        assert reco_df.index.isin(context_df.index).all(), "Recommendation index must be a subset of context index."
+        presc_actions_df = reco_df - context_df[constants.RECO_COLS]
+        presc_actions_df = presc_actions_df.rename(constants.RECO_MAP, axis=1)
+        presc_actions_df[constants.NO_CHANGE_COLS] = 0
+        context_actions_df = pd.concat([context_df[constants.CAO_MAPPING["context"]],
+                                            presc_actions_df[constants.CAO_MAPPING["actions"]]],
+                                            axis=1)
+        return context_actions_df
+
+    def _compute_percent_changed(self, context_actions_df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Calculates percent of land changed by prescriptor.
+        """
+        # Sum the positive diffs
+        percent_changed = context_actions_df[context_actions_df[constants.DIFF_LAND_USE_COLS] > 0][constants.DIFF_LAND_USE_COLS].sum(axis=1)
+        # Divide by sum of used land
+        total_land = context_actions_df[constants.LAND_USE_COLS].sum(axis=1)
+        total_land = total_land.replace(0, 1) # Avoid division by 0
+        percent_changed = percent_changed / total_land
+        change_df = pd.DataFrame(percent_changed, columns=["change"])
+        return change_df
+    
+    def _prescribe(self, candidate: Candidate, context_df=None) -> pd.DataFrame:
```

Comment:
Flow of prescription is: context df -> context tensor -> goes through model -> reco tensor -> reco df -> modify context df with reco df to get context + actions df

Reply:
Could also be a useful comment in the code or docstring